### PR TITLE
eliminate `grey` in prediction

### PR DESCRIPTION
**Pull request by hrj**
My mini benchmark results:

```
                       no-grey   signed IQ
-------------------------------------------
1_webp_ll.flif         78575        78560
2_webp_ll.flif         26591        26648
3_webp_ll.flif        156812       156806
5_webp_ll.flif         97864        98223
clock.flif             22538        22538
kodim07.flif          367234       370916
kodim08.flif          511741       512978
kodim12.flif          391329       391520
kodim13.flif          562924       565589
monkey.flif          1203177      1204173
pyani.flif            577229       583310
spinfox.flif          373800       373747
tea.flif              481203       481870
train.flif           1240191      1265475
watch_by8.flif        172195       172002
------------------------------------------
                     6263403      6304355

```

### CODE REVIEW

**Comment by hrj** [comment]
Tests are failing... don't merge.

I hadn't tested with `-n`. Fixing now.

**Comment by hrj** [comment]
Fixed for `-n`. However, I had to use grey value for the `0x0` pixel :(

Please check if the logic is correct (and also hopefully run the full benchmark suite on it).

Results for `-n`:

```
                       no-grey          master
---------------------------------------------
1_webp_ll.flif           77604          77441
2_webp_ll.flif           23939          24976
3_webp_ll.flif          153940         155812
5_webp_ll.flif           91875          93734
clock.flif               19971          20025
kodim07.flif            383419         384758
kodim08.flif            518223         524449
kodim12.flif            387141         395555
kodim13.flif            577146         579350
monkey.flif            1171094        1178192
pyani.flif              539312         565017
spinfox.flif            356668         358634
tea.flif                451153         451644
train.flif             1181076        1184211
watch_by8.flif          165360         166948
---------------------------------------------
                       6097921        6160746
```

**Comment by hrj** [comment]
Yikes, the build is still failing! Sorry...

File: src/common.cpp [hunk 1]
```diff
@@ -47,9 +45,9 @@ ColorVal predict_and_calcProps_scanlines(Properties &properties, const ColorRang
       }
       if (image.numPlanes()>3) properties[index++] = image(3,r,c);
     }
-    ColorVal left = (c>0 ? image(p,r,c-1) : grey[p]);;
-    ColorVal top = (r>0 ? image(p,r-1,c) : grey[p]);
-    ColorVal topleft = (r>0 && c>0 ? image(p,r-1,c-1) : grey[p]);
+    ColorVal left = (c>0 ? image(p,r,c-1) : (r > 0 ? image(p, r-1, c) : ((min + max) / 2)));
```
Comment (jonsneyers) [inline src/common.cpp]: Are min and max initialized? They are output args...

Comment (hrj) [inline src/common.cpp]: Yup, that was a mistake.

I have now tried two other fallbacks for `0,0` pixel: grey and min. Benchmark results:

```
Scanline method with reduced use of grey

                   fall-back=min   fallback=grey    orig
---------------------------------------------------------
1_webp_ll.flif           77604        77570         77441
2_webp_ll.flif           23939        23929         24976
3_webp_ll.flif          153940       153956        155812
5_webp_ll.flif           91875        91952         93734
clock.flif               19971        20086         20025
kodim07.flif            383602       383223        384758
kodim08.flif            518465       518439        524449
kodim12.flif            387185       387178        395555
kodim13.flif            576727       577313        579350
monkey.flif            1171108      1171155       1178192
pyani.flif              539312       539460        565017
spinfox.flif            356668       356597        358634
tea.flif                451153       451110        451644
train.flif             1181076      1180871       1184211
watch_by8.flif          165359       165379        166948
---------------------------------------------------------
                       6097984      6098218       6160746
```

`min` performs better than `grey` on average, and that's why I have used `min` and pushed the changes. Grey feels better but performs worse on the mini-benchmark. If you want me to change it to grey, let me know.

Travis-test is passing now.

Comment (hrj) [inline src/common.cpp]: `min` does have a conceptual advantage: it doesn't introduce a new color into the image, whereas `grey` can. Perhaps the ideal is to find a median color, from the available range of colors.

**Pull request merged**